packages/core-modules/contracts/mixins/AssociatedSystemsMixin.sol

```diff
@@ -0,0 +1,37 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/errors/InitError.sol";
+import "../storage/AssociatedSystemsStorage.sol";
+
+import "../interfaces/ITokenModule.sol";
+import "../interfaces/INftModule.sol";
+
+contract AssociatedSystemsMixin is AssociatedSystemsStorage {
+    bytes32 internal constant _KIND_ERC20 = "erc20";
+    bytes32 internal constant _KIND_ERC721 = "erc721";
+    bytes32 internal constant _KIND_UNMANAGED = "unmanaged";
+
+    function _getToken(bytes32 id) internal view returns (ITokenModule) {
+        _requireKind(id, _KIND_ERC20);
+        return ITokenModule(_associatedSystemsStore().satellites[id].proxy);
+    }
+
+    function _getNft(bytes32 id) internal view returns (INftModule) {
+        _requireKind(id, _KIND_ERC721);
+        return INftModule(_associatedSystemsStore().satellites[id].proxy);
+    }
+
+    function _requireKind(bytes32 id, bytes32 kind) internal view {
+        bytes32 actualKind = _associatedSystemsStore().satellites[id].kind;
```

Comment:
change it to emit a custom error

Reply:
https://github.com/Synthetixio/synthetix-v3/pull/1043/commits/3d1ab3d6afc720cf3de153c16727d2ce16356f7c